presto-main/src/main/java/com/facebook/presto/type/TinyintOperators.java

```diff
@@ -231,6 +233,13 @@ public static long hashCode(@SqlType(StandardTypes.TINYINT) long value)
         return TinyintType.hash((byte) value);
     }
 
+    @ScalarOperator(XX_HASH_64)
+    @SqlType(StandardTypes.BIGINT)
+    public static long xxHash64(@SqlType(StandardTypes.TINYINT) long value)
+    {
+        return XxHash64.hash((byte) value);
```

Comment:
We do not unpack it for all the inequality operators. Because of that i assume that the assumption is that the higher bits are always zero (or the same). Therefore i don't think the downcast to `byte` is needed here.